h2o-algos/src/main/java/hex/tree/dt/binning/BinningStrategy.java

```diff
@@ -38,51 +37,63 @@ public enum BinningStrategy {
             return roundToNDecimalPoints(number, DECIMALS_TO_CONSIDER);
         }
 
-        private List<Bin> createEmptyBinsFromBinningValues(List<Double> binningValues, double realMin, double realMax) {
-            List<Bin> emptyBins = new ArrayList<>();
+        private List<AbstractBin> createEmptyBinsFromBinningValues(List<Double> binningValues, double realMin, double realMax) {
+            List<AbstractBin> emptyBins = new ArrayList<>();
             // create bins between nearest binning values, don't create bin starting with the last value (on index size - 1)
             for (int i = 0; i < binningValues.size() - 1; i++) {
                 emptyBins.add(
-                        new Bin(roundToNDecimalPoints(binningValues.get(i)),
+                        new NumericBin(roundToNDecimalPoints(binningValues.get(i)),
                                 roundToNDecimalPoints(binningValues.get(i + 1))));
             }
             // set the firs min to some lower value (relative to step) so the actual value equal to min is not lost
-            emptyBins.get(0)._min = realMin - 0.0001 * (binningValues.get(1) - binningValues.get(0));
+            ((NumericBin) emptyBins.get(0))._min = realMin - 0.0001 * (binningValues.get(1) - binningValues.get(0));
```

Comment:
Here some setter would be nice. And 0.0001 could be constant, I am not sure about the name, just an idea. 

```suggestion
            ((NumericBin) emptyBins.get(0)).min(realMin - MIN_VALUE * (binningValues.get(1) - binningValues.get(0)));
```